### PR TITLE
fix: allow multiple colons in headers

### PR DESCRIPTION
**Pull request by boredland**
finding from https://github.com/remorses/genql/issues/74

### CODE REVIEW

File: cli/src/cli.ts [hunk 1]
```diff
@@ -33,13 +34,13 @@ const program = yargs
     .option('header', {
         alias: 'H',
         type: 'array',
-        // string: true,
```
Comment (boredland) [inline cli/src/cli.ts:36]: don't know why this was commented out. from what I see, validating scalars and headers as string makes sense.

File: cli/src/cli.ts [hunk 2]
```diff
@@ -119,22 +120,6 @@ generate(config)
         })
     })
 
-function parseColonSeparatedStrings(headersArray) {
```
Comment (boredland) [inline cli/src/cli.ts:122]: moved to a helper file to enable testing

**Comment by boredland** [comment]
@remorses did you have a look by any chance?

File: cli/src/cli.ts [hunk 3]
```diff
@@ -94,8 +94,8 @@ const config: Config = {
     useGet: program.get,
     schema: program.schema && readFile(program.schema),
     output: program.output,
-    headers: parseColonSeparatedStrings(program.header),
-    scalarTypes: parseColonSeparatedStrings(program.scalar),
+    headers: parseColonSeparatedStrings(program.header || []),
```
Comment (boredland) [inline cli/src/cli.ts:97]: did you try if this is needed? the function header of parseColonSeparatedStrings is:

```typescript
function parseColonSeparatedStrings(array: string[] = [])
```
that transpiles to 

```js
function parseColonSeparatedStrings(array) {
    if (array === void 0) { array = []; }
```
so... as I see it, you created duplicated code @remorses - or am I overseeing sth?

Comment (remorses) [reply cli/src/cli.ts:97]: You are probably right, the || just makes sure to handle null or “” values, I just want to be safe 😀

**Pull request merged**